api_app/script_analyzers/observable_analyzers/zoomeye.py

```diff
@@ -0,0 +1,107 @@
+import requests
+
+from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class ZoomEye(classes.ObservableAnalyzer):
+    base_url: str = "https://api.zoomeye.org/"
+
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get("api_key_name", "ZOOMEYE_KEY")
+        self.search_type = additional_config_params.get("search_type", "host")
+
+        # Setting query filters
+        self.query = {}
+        host_search_filters = [
+            "app",
+            "ver",
+            "device",
+            "os",
+            "service",
+            "cidr",
+            "port",
+            "city",
+            "country",
+            "asn",
+        ]
+        web_search_filters = [
+            "header",
+            "keywords",
+            "desc",
+            "title",
+            "site",
+        ]  # Included only which aren't included above
+        if "query" in additional_config_params:
```

Comment:
can you provide an example of how this `query` parameter works and what's the use case of it ?

Reply:
Other parameters like, City, Country, application can be passed in the query dict here. Actually, In ZoomEye docs, It's given as GET request parameter.

Reply:
Then, we should make `query` a string only and not an object since that is how it is also passed to the API. And in the docs, just specify the example against `query` parameter to be `port:21 city:beijing` (same one as given in zoomeye's docs).

Reply:
I felt, passing them as `dict` will be more comfortable. So, You say, Users directly pass the query string?